CityOfDetroit/COD-Design-System
# Support action buttons in UXDS

Part of #100 

## This PR

1. Creates a icon container for using icons within cards
2. Creates a higher-order element `<clickable/>` that causes child elements to behave as a clickable element
3. Adds funtionality to the card element to become a clickable component

## Alternatives

Instead of introducing `Clickable`, we could have:
1. Extended or re-used the `cod-button` component. The main reason to avoid this approach is that some elements should not be considered buttons for accessibility (e.g. a screenreader should treat action buttons as links).
2. Built another standalone component. The main reason to avoid this approach is that we miss out on composability of components (one of the main pros of a UXDS). E.g. we can re-use `<cod-clickable/>` and `<cod-card/>` when building arpa buttons.

## Testing

1. Build a couple stories that render clickable cards (styled as action buttons).
2. View them in storybook.

<img width="377" alt="Screenshot 2023-10-05 at 5 59 36 PM" src="https://github.com/CityOfDetroit/COD-Design-System/assets/143553259/3c755ba3-3742-48cb-a1a2-e02b636dab13">
<img width="400" alt="Screenshot 2023-10-05 at 5 59 32 PM" src="https://github.com/CityOfDetroit/COD-Design-System/assets/143553259/a09c5d84-d709-447b-a208-8a06e75a4a3b">
<img width="411" alt="Screenshot 2023-10-05 at 5 59 27 PM" src="https://github.com/CityOfDetroit/COD-Design-System/assets/143553259/11669399-4672-4978-9b7e-c1b8cedd4ed7">
<img width="398" alt="Screenshot 2023-10-05 at 5 59 17 PM" src="https://github.com/CityOfDetroit/COD-Design-System/assets/143553259/3c04e992-059c-4d7e-b0c5-74d882c6b41e">

Note: The action buttons are not equally sized in the test but can be by placing them in a card grid (like we do on the city homepage).

3. Observe cards in storybook to make sure there are no visual regressions.


<img width="404" alt="Screenshot 2023-10-05 at 6 11 41 PM" src="https://github.com/CityOfDetroit/COD-Design-System/assets/143553259/2edc8f2d-3153-413f-8c5f-92fb89b56ed6">
<img width="1052" alt="Screenshot 2023-10-05 at 6 11 37 PM" src="https://github.com/CityOfDetroit/COD-Design-System/assets/143553259/d589845f-3703-4436-90ed-dd36b509e2bf">

4. Run test suite

```
$ yarn test-storybook
...

Test Suites: 31 passed, 31 total
Tests:       157 passed, 157 total
Snapshots:   0 total
Time:        9.096 s
Ran all test suites.
```
